### DISCUSSION TITLE
Saving networks and checkpointing training.

### DISCUSSION THREAD

**Discussion by craftit** [neuronika/neuronika#76] [Q&A]
First, this looks like a fantastic library!    I am new to rust, so I may be miss understanding something, but I don't see any support for loading and saving networks and creating checkpoints during training?

**Comment by frjnn** [comment]
Hi, thanks for the interest in the project.

It should be possible to save and load models, and it should also be also possible to save and load optimisers. I say "should" because neuronika currently doesn't provide an API that does the job but is should be feasible given one. 

Anyways, for the latter, you should be able to serialise with serde the parameters of a model, then you could deserialise them and perform an element wise assignment between them and the ones of the model whose training you want to resume. 

Serialising and deserialising an optimizer would require to serialise its metadata, such as the last epoch, the momentum, the learning rate, ecc, which would be trivial and, also, to serialise the state of each parameter, which would be less trivial. To provide you with a more precise answer I must look into how serde handles references when deserialising.

As I said before, there are (almost totally) uncharted waters, but it can be done.

**Reply by craftit** [reply]
I will have a look, would you be interested in a pull request to make this a little easier?

**Reply by frjnn** [reply]
Sure, go ahead!
